backend/app/services/contest_service.py

```
@@ -48,11 +50,55 @@ def _build_response(contest: Contest, tally: dict, storage: StorageService) -> C
+
+    :param contest: The contest to check.
+    :param repo: Repository used to persist state changes.
+    """
+    now = datetime.now(UTC)
+
+    if contest.visibility == "scheduled":
+        sa = contest.scheduled_at
+        if sa is not None:
+            if sa.tzinfo is None:
+                from datetime import timezone
+                sa = sa.replace(tzinfo=timezone.utc)
```

Ruff (0.15.5)

[warning] Use `datetime.UTC` alias

Convert to `datetime.UTC` alias

(UP017)